src/layer/AtmosphereLayer.js

```diff
@@ -0,0 +1,414 @@
+
+            // Use this layer's night image when the light location is different than the eye location.
+            if (this.nightImageSource && this.lightLocation) {
+                this._activeTexture = dc.gpuResourceCache.resourceForKey(this.nightImageSource);
+                if (!this._activeTexture) {
+                    this._activeTexture = dc.gpuResourceCache.retrieveTexture(gl, this.nightImageSource);
+                }
+                textureBound = this._activeTexture && this._activeTexture.bind(dc);
+            }
+
+            for (var idx = 0, len = terrain.tessellator.currentTiles.length; idx < len; idx++) {
+                var currentTile = terrain.tessellator.currentTiles.tileArray[idx];
+                // Use the vertex origin for the terrain tile.
+                var terrainOrigin = currentTile.referencePoint;
+                program.loadVertexOrigin(gl, terrainOrigin);
+
+                // Use the draw context's modelview projection matrix, transformed to the tile's local coordinates.
+                this._mvpMatrix.copy(dc.navigatorState.modelviewProjection);
+                this._mvpMatrix.multiplyByTranslation(terrainOrigin[0], terrainOrigin[1], terrainOrigin[2]);
+                program.loadModelviewProjection(gl, this._mvpMatrix);
+
+                // Use a tex coord matrix that registers the night texture correctly on each terrain.
+                if (textureBound) {
+                    this._texMatrix.setToUnitYFlip();
+                    this._texMatrix.multiplyByTileTransform(currentTile.sector, this._fullSphereSector);
+                    program.loadTexMatrix(gl, this._texMatrix);
+                }
+
+                // Draw the tile, multiplying the current fragment color by the program's secondary color.
+                program.loadFragMode(gl, program.FRAGMODE_GROUND_SECONDARY);
+                gl.blendFunc(gl.DST_COLOR, gl.ZERO);
+                terrain.tessellator.beginRendering(dc);
+                terrain.beginRenderingTile(dc, currentTile);
+                terrain.renderTile(dc, currentTile);
+                terrain.endRenderingTile(dc, currentTile);
+                terrain.tessellator.endRendering(dc);
+
+                // Draw the terrain as triangles, adding the current fragment color to the program's primary color.
+                var fragMode = textureBound ?
+                    program.FRAGMODE_GROUND_PRIMARY_TEX_BLEND : program.FRAGMODE_GROUND_PRIMARY;
+                program.loadFragMode(gl, fragMode);
+                gl.blendFunc(gl.ONE, gl.ONE);
+                terrain.tessellator.beginRendering(dc);
+                terrain.beginRenderingTile(dc, currentTile);
+                terrain.renderTile(dc, currentTile);
+                terrain.endRenderingTile(dc, currentTile);
+                terrain.tessellator.endRendering(dc);
+            }
+
+            // Restore the default World Wind OpenGL state.
+            gl.blendFunc(gl.ONE, gl.ONE_MINUS_SRC_ALPHA);
```

Comment:
See the comment above regarding program.vertexPointLocation and program.vertexTexCoordLocation.